app/src/main/java/ch/heigvd/pro/b04/android/Home/Home.java

```diff
@@ -34,13 +36,34 @@ public int getSpanSize(int position) {
             }
         });
 
-        EmojiAdapter emojiGridAdapter = new EmojiAdapter(state, this);
-        emojiGrid.setAdapter(emojiGridAdapter);
+        EmojiAdapter emojiAdapter = new EmojiAdapter(state, this);
+        emojiGrid.setAdapter(emojiAdapter);
         emojiGrid.setLayoutManager(manager);
 
+        // Code of selected emojis
+        RecyclerView emojiCode = findViewById(R.id.home_emoji_code);
+        GridLayoutManager llm = new GridLayoutManager(this, 4);
+        //llm.setOrientation(RecyclerView.HORIZONTAL);
+        manager.setSpanSizeLookup(new GridLayoutManager.SpanSizeLookup() {
+            @Override
+            public int getSpanSize(int position) {
+                return position == 0 ? 4 : 1;
+            }
+        });
+        EmojiCodeAdapter emojiCodeAdapter = new EmojiCodeAdapter(state, this);
+        emojiCode.setAdapter(emojiCodeAdapter);
+        emojiCode.setLayoutManager(llm);
+
         state.getToken().observe(this, token -> {
-            Intent intent = new Intent(this, PollActivity.class);
-            startActivity(intent);
+            if (token.getToken() == "Error") {
+                // TODO : Make it a bit prettier if possible, but at least, it does the job
+                Toast toast = Toast.makeText(this, "Wrong code !", Toast.LENGTH_LONG);
+                toast.setGravity(0, 0, 0);
+                toast.show();
```

Comment:
Doesn't work

Reply:
It actually does, don't know why it doesn't on your end...